example-app/App.tsx

```diff
@@ -40,6 +40,41 @@ const Header = () => {
 };
 
 const App = () => {
+  const [result, setResult] = React.useState<{
+    success: boolean | undefined;
+    errorReason: string | undefined;
+  }>({success: undefined, errorReason: undefined});
```

Comment:
```suggestion
  const [result, setResult] = React.useState<{
    success: boolean;
    errorReason: string | undefined;
  } | undefined>(undefined);
```

Maybe easier? Now you can see if the object exists, then it has a success value.

Reply:
Went fancy way. Long signatures, rows and rows, but quite neat.